Subject: [PATCH] Fix bugs regarding calculate dalvik-cache size

Symptom:
App size measure too large.

Root cause:
A/B Update mounts system_other(system_b) and copy prebuilt odex
files from it to data/dalvik-cache in the first bootng.
this working is executed by root so after first boot, the oat files have root uid and gid.

In order to calculte app size, StorageStatsService call
queryStatsForPackage method and if the pacakge is multiple package
which have shared UID and SystemUID, appId is '0' as USER_SYSTEM.
As a result, when calculate dalvik, the size is measure of all files
for root gid

Solution:
Consider root_gid for shared_gid when calculate dalvik-cache for appsize
Test: manual - checked app size in setting

diff --git a/cmds/installd/InstalldNativeService.cpp b/cmds/installd/InstalldNativeService.cpp
index b0661c5..1f92a33 100644
--- a/cmds/installd/InstalldNativeService.cpp
+++ b/cmds/installd/InstalldNativeService.cpp
@@ -1403,7 +1403,7 @@
         if (!uuid) {
             ATRACE_BEGIN("dalvik");
             int32_t sharedGid = multiuser_get_shared_gid(0, appId);
-            if (sharedGid != -1) {
+            if (sharedGid != -1 && sharedGid != 0) {
                 calculate_tree_size(create_data_dalvik_cache_path(), &stats.codeSize,
                         sharedGid, -1);
             }
